src/core/include/openvino/core/model.hpp

```diff
@@ -24,9 +24,15 @@
 #include "openvino/runtime/tensor.hpp"
 
 namespace ov {
+namespace frontend {
+class FrontEnd;
+}
 class ModelAccessor;
 /// A user-defined function.
 class OPENVINO_API Model : public std::enable_shared_from_this<Model> {
+    friend class ov::frontend::FrontEnd;
+    std::shared_ptr<void> m_shared_object;  // Frontend plugin shared object handle.
+
```

Comment:
Btw, should not we also fix a function that clones current ov::Model?

Reply:
Yes we should

Reply:
Fixed